src/eslint/react-server-components-directive.js

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import micromatch from "micromatch";
+import path from "node:path";
+
+export default {
+  meta: {
+    fixable: "code",
+    schema: {
+      type: "array",
+      items: {
+        type: "object",
+        properties: {
+          entryFilesPattern: {
+            type: "string",
+          },
+        },
+      },
+    },
+  },
+  create(context) {
+    const entryFilesPattern = path.resolve(context.options[0]?.entryFilesPattern ?? "./src/*/index.{ts,tsx}");
+    const filename = path.resolve(context.filename);
+    const isEntryFile = micromatch.isMatch(filename, entryFilesPattern);
```

Comment:
This resembles how our other custom rule works: https://github.com/cloudscape-design/components/blob/main/build-tools/eslint/ban-files.js

eventually, we need to move that one in this package as well